test/jdk/java/nio/file/WatchService/LotsOfEntries.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2010, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 8330077
+ * @summary Tests WatchService behavior with more entries in a watched directory
+ *     than the default event limit
+ * @library ..
+ * @run main/othervm LotsOfEntries 600 fail
+ * @run main/othervm -Djdk.nio.file.WatchService.maxEventsPerPoll=invalid LotsOfEntries 600 fail
+ * @run main/othervm -Djdk.nio.file.WatchService.maxEventsPerPoll=-5 LotsOfEntries 5 fail
+ * @run main/othervm -Djdk.nio.file.WatchService.maxEventsPerPoll=5 LotsOfEntries 5 pass
```

Comment:
I ran 100 repeats of this test on each of four different platforms. 398 executions passed, but two failed, both in the sub-test at line 32. I do not know what happened in these two cases for the sub-tests at lines 33-35 as these were not run. The output was like:

```
jtreg_open_test_jdk_java_nio_file_WatchService_LotsOfEntries_java/tmp/name7869734652501521174/entry1]
	at LotsOfEntries.testCreateLotsOfEntries(LotsOfEntries.java:84)
	at LotsOfEntries.main(LotsOfEntries.java:96)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.sun.javatest.regtest.agent.MainWrapper$MainTask.run(MainWrapper.java:138)
	at java.base/java.lang.Thread.run(Thread.java:1575)

JavaTest Message: Test threw exception: java.lang.RuntimeException: Expected events on: [multiple flle names omitted]
```

Reply:
I slightly lowered/raised the numbers of files to be created. We can tune this more if needed, I don't think fully precise control over the number of events is required.

Reply:
We just don't want to have random failures creating noise. Tests are run many times each day. I think the problematic line here is
```
 32  * @run main/othervm -Djdk.nio.file.WatchService.maxEventsPerPoll=5 LotsOfEntries 5 pass
```
This failed I think in only 3 of 300 repeats on linux-aarch64.

Reply:
Yes what I meant is that the numbers in the test are essentially arbitrary. Now that they are further away from the limit, I would expect failures to be less likely. But if there are still failures, I can make the difference larger.

Reply:
It would be good to know why the failures happened so they can be prevented.

Reply:
Could you share the omitted list of files? The error message should give us the diff, which would be helpful in figuring out what went wrong.

I don't have access to a linux-aarch64 machine, but I will also try to stress test this test.

Reply:
I pushed a new change that reverts to the original test cases, but adds more debug output. @bplb Could you perhaps run the test again on linux-aarch64 now that we will get to see all events that were sent?

Reply:
> Could you share the omitted list of files? The error message should give us the diff, which would be helpful in figuring out what went wrong.

Voila:
```
java.lang.RuntimeException: Expected events on: [entry1, entry0, entry3, entry2, entry4], got: [entry1, entry0]
        at LotsOfEntries.testCreateLotsOfEntries(LotsOfEntries.java:84)
        at LotsOfEntries.main(LotsOfEntries.java:96)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at com.sun.javatest.regtest.agent.MainWrapper$MainTask.run(MainWrapper.java:138)
        at java.base/java.lang.Thread.run(Thread.java:1575)
```

Reply:
> Could you perhaps run the test again on linux-aarch64 now that we will get to see all events that were sent?

@fmeum Will initiate in the next few minutes. Thanks.

Reply:
I am seeing a build error now:
```
jdk/open/src/java.base/share/classes/java/nio/file/WatchService.java:105: error: reference not found
 * {@link StandardWatchEventKind#OVERFLOW OVERFLOW} event is queued. This
          ^
1 error
```

Reply:
`StandardWatchEventKind` needs to be `StandardWatchEventKinds` (plural).

Reply:
Thanks, fixed the typo.

Reply:
Running tests now with locally fixed typo.

Reply:
Still seeing failures
```
;; This buffer is for text that is not saved, and for Lisp evaluation.
;; To create a file, visit it with C-x C-f and enter text in its buffer.

#section:main
----------messages:(6/343)----------
command: main -Djdk.nio.file.WatchService.maxEventsPerPoll=-5 LotsOfEntries 5 fail
reason: User specified action: run main/othervm -Djdk.nio.file.WatchService.maxEventsPerPoll=-5 LotsOfEntries 5 fail 
started: Fri Apr 19 16:27:35 UTC 2024
Mode: othervm [/othervm specified]
finished: Fri Apr 19 16:27:35 UTC 2024
elapsed time (seconds): 0.233
----------configuration:(0/0)----------
----------System.out:(4/418)----------
register name13539591746028431398 for events
create 5 entries
poll watcher...
poll expecting overflow...
----------System.err:(12/689)----------
java.lang.RuntimeException: Expected overflow event
	at LotsOfEntries.testCreateLotsOfEntries(LotsOfEntries.java:74)
	at LotsOfEntries.main(LotsOfEntries.java:104)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at com.sun.javatest.regtest.agent.MainWrapper$MainTask.run(MainWrapper.java:138)
	at java.base/java.lang.Thread.run(Thread.java:1575)

JavaTest Message: Test threw exception: java.lang.RuntimeException: Expected overflow event
```